_posts/2018-11-27-pool-features.md

```diff
@@ -0,0 +1,149 @@
+
+Starting from [release 1.5][release-1.5] set of new features have been introduced. It is **crucial for data safety** to know if e.g. [Reliability, Availability and Serviceability \(RAS\)][RAS] features are enabled in the pool or not. To meet the need bunch of new possibilities are now available:
+
+* `pmempool` tool [commands for managing features][pmempool-feature.1]
+* *libpmempool* [API for managing features][pmempool-feature.3]
+* CTL namespace for [setting a `SHUTDOWN_STATE` initial value](#set-shutdown_state-initial-value)
+
+# The features taxonomy
+
+Any feature can be disabled or enabled in the pool and supported or not supported in the used version of the PMDK library. All features are divided into three groups:
+
+* compatible features:
+	* `CHECK_BAD_BLOCKS`
+* incompatible features:
+	* `SINGLEHDR`
+	* `CKSUM_2K`
+	* `SHUTDOWN_STATE`
+* read-only compatible features:
+	* \(empty group\)
+
+This partition gives the following matrix:
+
+<table>
+	<thead>
+		<tr>
+			<th colspan="2" rowspan="2">pool feature</th>
+			<th colspan="2">PMDK library support</th>
+		</tr>
+		<tr>
+			<th>yes</th>
+			<th>no</th>
+		</tr>
+	</thead>
+	<tbody>
+		<tr>
+			<th rowspan="2">compat</th>
+			<th>enabled</th>
+			<td>(1)</td>
+			<td>the pool can be opened and used but the feature has no effect</td>
+		</tr>
+		<tr>
+			<th>disabled</th>
+			<td>(2)</td>
+			<td>(2)</td>
+		</tr>
+		<tr>
+			<th rowspan="2">incompat</th>
+			<th>enabled</th>
+			<td>(1)</td>
+			<td>the pool can <strong>NOT</strong> be opened and used because PMDK has to support the feature to handle the pool properly</td>
+		</tr>
+		<tr>
+			<th>disabled</th>
+			<td>(2)</td>
+			<td>(2)</td>
+		</tr>
+		<tr>
+			<th rowspan="2">ro&nbsp;compat</th>
+			<th>enabled</th>
+			<td>(1)</td>
+			<td>the pool can be opened but it is opened in the <strong>read-only mode</strong></td>
+		</tr>
+		<tr>
+			<th>disabled</th>
+			<td>(2)</td>
+			<td>(2)</td>
+		</tr>
+	</tbody>
+</table>
+
+- (1) If the feature is enabled and supported by the used PMDK library it has effect as described in [the documentation][pmempool-feature.3].
+- (2) If the feature is disabled it does not matter if it is supported or not. It has no effect.
+
+## `CKSUM_2K` dependant features
+
+An initial 4K of a pool is reserved for a pool header. The pool header is checksummed to control its consistency. If `CKSUM_2K` feature is enabled only first 2K of the pool header is checksummed otherwise a whole 4K. When only a first 2K of a pool header is checksummed the other 2K can be used used to store more dynamically changing data without affecting pool header consistency. Currently, the only feature which makes use of this extra space is `SHUTDOWN_STATE` feature.
```

Comment:
The first 4k of a pool...
to ensure its consistency
... feature is enabled, only the first 2k of the pool...

When only first ... -> "This feature allows the library more flexibility, as it's required for features that require extra pool header capacity. Currently, that's only `SHUTDOWN_STATE`"